pkg/daemon/daemon.go

```diff
@@ -1583,11 +1587,62 @@ func PersistNetworkInterfaces(osRoot string) error {
 		if err := cmd.Run(); err != nil {
 			return fmt.Errorf("failed to run nmstatectl: %w", err)
 		}
+	} else if hostos.IsEL9() {
+		ifnames, err := getIfnamesFromLinkFiles(osRoot)
+		if err != nil {
+			glog.Warningf("Failed to persist NIC names: %v", err)
+			return nil
+		}
+
+		ifnamesKeys := []string{}
+		for ifname := range ifnames {
+			ifnamesKeys = append(ifnamesKeys, ifname)
+		}
+		if len(ifnamesKeys) > 0 {
+			logSystem("Have persisted ifnames for %s", strings.Join(ifnamesKeys, ", "))
+		}
 	}
 
 	return nil
 }
 
+// getIfnamesFromLinkFiles scans /etc/systemd/network for nmstate link files and
+// extracts the pinned network interface names and MAC addresses.
+func getIfnamesFromLinkFiles(osRoot string) (map[string]string, error) {
```

Comment:
I'd hoped to avoid having the MCO do more than simply invoke nmstate.  It's not clear to me at all that it's a long term win to have this logic here versus having nmstate continue to have `--inspect`.

Reply:
That code was added as part of this PR for helping with inspection, but its main purpose was first as part of https://github.com/openshift/machine-config-operator/pull/3684 to know which kargs to append and remove. Not against moving it to nmstate (e.g. one idea I had there was to have the stamp file contain JSON of the system state instead), but I think the bigger complexity cost come from the karg handling. And there it's less obvious to me whether nmstate should modify rpm-ostree state too. It could, but because the MCO is tasked with owning node state and rpm-ostree calls (and e.g. cleans up deployments), it'd be easier to follow I think if it was kept in the MCO.

Reply:
OK I missed the kargs bit.  But um...are you sure that even works?  Because today the MCO always removes the pending deployment before doing an OS update.

Reply:
> OK I missed the kargs bit. But um...are you sure that even works? Because today the MCO always removes the pending deployment before doing an OS update.

Yup, that's why I did [`6d07862` (#3684)](https://github.com/openshift/machine-config-operator/pull/3684/commits/6d0786228ef0cc2d5b0d8e6bab25b8664ef97afc). 

You can see it working in https://github.com/openshift/machine-config-operator/pull/3684#issuecomment-1518653633.